conans/tools.py

```diff
@@ -134,4 +133,6 @@ def patch(base_path=None, patch_file=None, patch_string=None):
     else:
         patchset = fromstring(patch_string.encode())
 
-    patchset.apply(root=base_path)
+    if not patchset.apply(root=base_path):
+        raise Exception("Failed to apply patch: %s" % patch_file)
```

Comment:
Would you mind changing it to a ConanException?
ConanExceptions are more controlled, produced a cleaner output (without a traceback that is ugly for most users), as we know it is something that conan produced. On the contrary regular Exceptions are non foreseen problems, bugs, and other issues, and even if we capture them (eg. in ``Conanfile.config()`` method, to allow the package creators to raise regular Exceptions) in some contexts, the usual is that they produce an unformatted stack trace dump, that by design is considered always a conan bug.
Makes sense this design criteria?